File: Exercises/rb101-rb109_small_problems/easy_8/leading_substrings.rb
# frozen_string_literal: true

def leading_substrings(string)
  arr = []
  0.upto(string.size - 1) do |num|
    arr << string[0..num]
  end
  arr
end

puts leading_substrings('abc') == ['a', 'ab', 'abc']
puts leading_substrings('a') == ['a']
puts leading_substrings('xyzzy') == ['x', 'xy', 'xyz', 'xyzz', 'xyzzy']

File: ls215/exercises/small_problems/list_processing/leading_substrings.js
function leadingSubstrings(string) {
  const chars = string.split('');

  return chars.map((char, index) => {
    return chars.slice(0, index + 1).join('');
  });
}

console.log(leadingSubstrings('abc'));      // ["a", "ab", "abc"]
console.log(leadingSubstrings('a'));        // ["a"]
console.log(leadingSubstrings('xyzzy'));    // ["x", "xy", "xyz", "xyzz", "xyzzy"]
